### DISCUSSION TITLE
Will Compose MP's Lifecycle and ViewModel implementations be adopted here?

### DISCUSSION THREAD

**Discussion by har-nick** [adrielcafe/voyager#401] [General]
Saw they were adding and improving their impls on their repo, and I was just wondering how it'd affect the development of Voyager, and whether we might need to migrate in the future.

**Comment by Syer10** [comment]
Once its ready, it should be simple to convert the Android-specific implementation of the Voyager ViewModelStore and Lifecycle to the multi-platform variants.

ScreenModels won't go away since they are easily tied to Voyager-specific things, while ViewModels have a stricter implementation.

**Reply by DevSrSouza** [reply]
That is true, but currently the viewmodel-compose package KMP depends on `compose-ui`, this is breaks one of the Voyager important aspects from the Navigator and Core: It is Compose Runtime based, not compose ui, in practice this means that you could for example write a extension for Redwood. Currently this is not possible.
